### ISSUE TITLE
All content read as expanded in walkthrough for SR users

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#225977]
> Here is how you can reproduce it.
> 
> 
> 
> 1. Open VS code on Windows and make sure NVDA is running.
> 
> 2. In the command palette, run "Welcome - Open Walkthrough.".
> 
> 3. Press the letter B to navigate by buttons using the browse cursor. Observe that nVDA announces the buttons and corresponding checkboxes as a single control. (   check box  not checked    Sync settings across devices  button  collapsed).
> 
> 4. Use letter X to navigate by checkboxes using the Browse cursor. Observe that NVDA just reads unlabeled checkboxes. (check box  not checked    button  collapsed ).
> 
> 5. locate the button "Tune your settings", and press enter to expand the content.
> 
> 6. Use the down arrow key to read the expanded content.
> 
> 
> 
> At this point, every single string of text in the expanded content is read as an expanded button, so it is not possible to tell what is actionable except by context. (button  expanded    Customize every aspect of VS Code and your extensions to your liking. Commonly used settings are). Also, it seems like it is not possible to collapse an element once expanded using the keyboard or the screen reader. These are usually signs of improper nesting of tags. 

 _Originally posted by @rperez030 in [#223787](https://github.com/microsoft/vscode/issues/223787#issuecomment-2293769695)_

**Comment by meganrogge** [comment]
I just met with @jooyoungseo. As @rperez030 discovered, our use of the tree view for this is confusing for screen reader users. From their perspective, the point of a tree view is to have a hierarchy of collapsible/expandable elements. 

We auto-expand the element when it becomes focused. 

We're going to think about a custom editor for the accessibility help info instead.

**Comment by rperez030** [comment]
> I just met with @jooyoungseo. As @rperez030 discovered, our use of the tree view for this is confusing for screen reader users. From their perspective, the point of a tree view is to have a hierarchy of collapsible/expandable elements.

This is not really a treeview, but an accordion pattern. If implemented properly, it could result in a good user experience. a sample implementation not intended for production can be found at [https://www.w3.org/WAI/ARIA/apg/patterns/accordion/](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/).

> 
> We auto-expand the element when it becomes focused.

That can create an accessibility problem not only for screen reader users, but for sighted keyboard users as well. Imagine someone trying to use the Tab key or an alternative input device to navigate the walk-through, and having all elements expanding automatically as soon as they receive focus. That defeats the purpose of having collapsible elements in the first place.
> 
> We're going to think about a custom editor for the accessibility walkthrough instead of this.

Would you consider improving this one instead. Like I said, it can create obstacles for more than screen reader users.

**Comment by meganrogge** [comment]
Yes @rperez030, we are on the same wavelength 😄. We will fix this one. Thanks.

https://github.com/microsoft/vscode/issues/223787#issuecomment-2310689515

**Comment by bhavyaus** [comment]
Also, just tried out the default narrator on windows and voice over on the mac and see only the currently selected step description and content being read out and correctly labelled as `expanded`.  Could you help clarify the issue being described here? I'm not entirely sure I understand the problem.

**Comment by meganrogge** [comment]
Hi @bhavyaus, this only reproduces on windows with NVDA

**Comment by bhavyaus** [comment]
I'm not sure how to fix accessibility bugs specific to NVDA without breaking the default accessibility apps on windows and mac. 
Also tried out other UX in our product where we have both text and buttons in the content description (example: Welcome View when no workspace is open) and I see the same behavior where all of the expanded content is read.

This feels like an issue we need to fix across the various UX .

**Comment by meganrogge** [comment]
@bhavyaus one idea would be when in screen reader mode to not auto-expand

**Comment by meganrogge** [comment]
@bhavyaus should this be labeled `wont-fix`?

**Comment by roblourens** [comment]
Taking it out of the verification query

**Comment by meganrogge** [comment]
@rperez030 thank you for clarifying. 

@bhavyaus, could we please revisit this? LMK if you want help.

**Comment by bhavyaus** [comment]
Apologies but will get this fixed up in April.  And all the suggestions mentioned above make sense.
